### PR TITLE
New unit tests

### PR DESCRIPTION
**Pull request by tisnik**

### CODE REVIEW

File: tests/test_directed_graph.py [hunk 1]
```diff
@@ -60,6 +122,74 @@ def test_add_edge():
...
+
+
+def test_add_edge_between_nonexistent_vertices():
+    """Test whether method DirectedGraph.add_edge() works correctly."""
+    g = DirectedGraph()
+    v0 = Vertex(0, {})
+    v1 = Vertex(1, {})
+
+    with pytest.raises(Exception):
+        g.add_edge(from_id=v0.id, to_id=v1.id)
```
Comment (jpopelka) [inline tests/test_directed_graph.py]: 2 questions:
- why does it not work ? (maybe add a comment there)
- can the exception be more specific than generic `Exception` ?

File: tests/test_directed_graph.py [hunk 2]
```diff
@@ -74,6 +204,22 @@ def test_find_vertex():
...
+    list_vertices = DirectedGraph.find_common_reachable_vertices(input_vertices=None)
+    assert list_vertices is None
+
+    list_vertices = DirectedGraph.find_common_reachable_vertices(input_vertices=[])
+    assert list_vertices is None
+
+    v0 = Vertex(0, {})
+    list_vertices = DirectedGraph.find_common_reachable_vertices(input_vertices=[v0])
+    assert list_vertices is not None
+    assert list_vertices[0] == v0
```
Comment (jpopelka) [inline tests/test_directed_graph.py]: the variable `g` seems to be unused

Comment (tisnik) [reply tests/test_directed_graph.py]: yes the tested method is static and I left the variable here. Fixing it. TYVM!

File: tests/test_directed_graph.py (hunk 1 above)
Comment (tisnik) [reply tests/test_directed_graph.py]: the vertexes are simply not part of the directed graph, so they can't be find in the internal dictionary with all vertexes. I'll add comment.

**Comment by tisnik** [comment]
[test]

**Pull request merged**